onnxruntime/core/optimizer/propagate_cast_ops.h

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/optimizer/graph_transformer.h"
+
+namespace onnxruntime {
+
+/**
+@Class PropagateCastOps
+
+Propagate FP16 Cast operations up the graph and FP32 Cast operations down the graph
+
+*/
+class PropagateCastOps : public GraphTransformer {
```

Comment:
>PropagateCastOps [](start = 6, length = 16)

should this be renamed since this is transformation?

Reply:
what is the naming convention?

Reply:
How about CastPropagation?